src/java_bytecode/java_static_initializers.cpp

```diff
@@ -39,6 +70,37 @@ bool is_clinit_wrapper_function(const irep_idt &function_id)
   return has_suffix(id2string(function_id), clinit_wrapper_suffix);
 }
 
+/// Add a new symbol to the symbol table.
+/// Note: assumes that a symbol with this name does not exist.
+/// /param name: name of the symbol to be generated
+/// /param type: type of the symbol to be generated
+/// /param value: initial value of the symbol to be generated
+/// /param is_thread_local: if true this symbol will be set as thread local
+/// /param is_static_lifetime: if true this symbol will be set as static
+/// /return returns new symbol.
+static symbolt add_new_symbol(
```

Comment:
Should probably rename `add_new_variable_symbol`, since the `is_lvalue` and `is_state_var` choices restrict it to vars (not types, not functions)